### DISCUSSION TITLE
Missing ports on device

### DISCUSSION THREAD

**Discussion by PeterKnotek** [netbox-community/netbox#9440] [Q&A]
Hi All
After change device type in device setting.
Miing all ports in devices
- remove device type : not work

Only vork create new device and old device delete ...

**Comment by matejv** [comment]
This is by design. Netbox only populates device interfaces and other components from selected device type when you first create a device. It will not make changes to device components if you change device type later.

There is a script that could work for you: [add_device_type_components.py](https://github.com/netbox-community/reports/blob/master/scripts/add_device_type_components.py). Run it after you change device type of a device. Note it will only create new components. You will have to remove any that are redundant yourself.
